api/napi/__test__/compiler.spec.ts

```diff
@@ -237,3 +238,35 @@ test('non-existent properties and callbacks', (t) => {
   t.is(callback_err!.code, 'GenericFailure');
   t.is(callback_err!.message, 'Callback non-existent-callback not found in the component');
 })
+
+test('Window constructor', (t) => {
```

Comment:
These two tests look a little ... out of place in the `compiler.spec.ts`. They're not really compiler related IMO. I suggest to move them into a separate file.